### DISCUSSION TITLE
transferFrom wrong~!!!!

### DISCUSSION THREAD

**Discussion by jerryking0421** [lessmore92/PHPEthereumERC20#8] [Q&A]
use transferFrom function,api respose content:  Uncaught RuntimeException: insufficient funds for gas * price + value
but spender's account have eth

**Comment by nathanielajayi7** [comment]
The problem is the gas limits and gas prices, you have to set this params to custom because the existing ones resolve to a large amount for supposed fee.

**Reply by jerryking0421** [reply]
Thank U nathanielajayi7 
1.how to set custom
2. I changed price and push transferFrom order , but the order  always pending  fail

**Reply by nathanielajayi7** [reply]
https://github.com/lessmore92/php-erc20/blob/main/src/Foundation/StandardERC20Token.php

Visit this file, you can see how the TransactionBuilder class was used. The gasPrice and gasLimits are in hexadecimal

**Comment by suoten** [comment]
I am also the same mistakes, but don't know how specific custom. Is to modify the value of the protected $gasLimits

**Comment by suoten** [comment]
> use transferFrom function,api respose content: Uncaught RuntimeException: insufficient funds for gas * price + value but spender's account have eth

I also encountered the same problem, can you tell me you solved?

**Comment by lessmore92** [comment]
Thank you all,
I made an update (`v1.0.1`), in this version, you can set your custom `gasPrice` and `gasLimit`.

As you can see, I added two params with default values, you can set your own value or leave default values.
`transferFrom(string $spender, string $from, string $to, float $amount, string $gasLimit = 'default', string $gasPrice = 'default')`

Please update to the latest version.

**Reply by jerryking0421** [reply]
Thank U,I also have question, Do we need the spender param when using the transferFrom function.  I see examples of other people's homes, just spender's key.
